### PR TITLE
Fix last one hardcoded unversioned python command

### PR DESCRIPTION
**Pull request by frenzymadness**
I'm sorry, it seems that I missed this one. I've prepared this PR in a browser so please test it before merging.

### CODE REVIEW

**Comment by frenzymadness** [comment]
I'd like to debug the issue but I don't have an access to a machine with windows. But it seems that it cannot handle the full path to executable. It might make sense to omit the call to which if the first part of the command is an absolute path.

**Comment by afshin** [comment]
>  It might make sense to omit the call to which if the first part of the command is an absolute path.

This sounds reasonable 👍

**Comment by frenzymadness** [comment]
I might try to implement it but I don't have a windows machine to test it.

**Comment by frenzymadness** [comment]
Implemented in the last commit.

File: jupyter_packaging/setupbase.py [hunk 1]
```diff
@@ -259,12 +259,14 @@ def run(cmd, **kwargs):
     kwargs.setdefault('shell', os.name == 'nt')
     if not isinstance(cmd, (list, tuple)):
         cmd = shlex.split(cmd)
-    cmd_path = which(cmd[0])
-    if not cmd_path:
-        raise ValueError("Aborting. Could not find cmd (%s) in path. "
-                 "If command is not expected to be in user's path, "
-                 "use an absolute path." % cmd[0])
-    cmd[0] = cmd_path
+    if not os.path.isabs(cmd[0]):
```
Comment (afshin) [inline jupyter_packaging/setupbase.py:262]: Maybe try `str(cmd[0]).replace(os.sep, “/“)` for the path to test.

Comment (frenzymadness) [reply jupyter_packaging/setupbase.py:262]: Could you please explain why should I replace the path separator? `sys.executable` should be an absolute path to the interpreter and `os.path.isabs` should support windows paths.
I don't have an access to windows to debug it.

**Comment by afshin** [comment]
It looks like the path fix needs to be on `sys.executable` itself. You shouldn't need any changes to `setupbase.py` if you make only these changes:
```diff
diff --git a/tests/test_utility_functions.py b/tests/test_utility_functions.py
index 334c085..ea8127e 100644
--- a/tests/test_utility_functions.py
+++ b/tests/test_utility_functions.py
@@ -1,4 +1,5 @@
-
+import os
+import sys
 from unittest.mock import patch
 import pytest

@@ -28,7 +29,8 @@ def test_combine_commands():


 def test_run():
-    assert pkg.run('python --version') == 0
+    python = sys.executable.replace(os.sep, '/')
+    assert pkg.run(f'{python} --version') == 0

     with pytest.raises(ValueError):
         pkg.run('foobarbaz')
```

**Comment by frenzymadness** [comment]
> It looks like the path fix needs to be on `sys.executable` itself. You shouldn't need any changes to `setupbase.py` if you make only these changes:

But I still don't know why should I change the path separators from Windows to Linux. I'd say that windows paths should work on windows.

**Comment by afshin** [comment]
In this case the limitation is `shutil.which` not handling the Windows path, part of the standard library.  Windows support is best-effort, since none of the core devs are Windows users.

**Comment by vidartf** [comment]
> Windows support is best-effort, since none of the core devs are Windows users.

👋 

I added a commit to the PR to resolve the issue: It was using shlex.split which seems to need a `posix=False` on Windows to not bork windows paths.

**Comment by blink1073** [comment]
Thanks @frenzymadness and @vidartf!

**Pull request merged**